internal/ingress/annotations/conditions/types.go

```diff
@@ -0,0 +1,255 @@
+package conditions
+
+import (
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/internal/aws"
+	"github.com/pkg/errors"
+)
+
+const (
+	FieldHostHeader        = "host-header"
+	FieldPathPattern       = "path-pattern"
+	FieldHTTPHeader        = "http-header"
+	FieldHTTPRequestMethod = "http-request-method"
+	FieldQueryString       = "query-string"
+	FieldSourceIP          = "source-ip"
+)
+
+// Information about a host header condition.
+type HostHeaderConditionConfig struct {
+	// One or more host names. The maximum size of each name is 128 characters.
```

Comment:
Nit: Should it say "maximum size" or "maximum length"?

Reply:
Aha, both applies a guess :D.
In fact, the struct and comments is directly copied from AWSSDKGo. (instead of using it directly, i create my own copy to 
1. avoid been broken when upgrades SDK
2. add extensions, like allow ServiceName/ServicePort instead of TargetGroupArn in forwardConfig
)